hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1023,6 +1023,9 @@ def as_strategy(strat_or_callable, thing, final=True):
                 "strings."
             )
         raise InvalidArgument(f"thing={thing!r} must be a type")  # pragma: no cover
+    if types.is_forbidden_to_dispatch(thing):  # pragma: no cover
+        # Code like `st.from_type(TypeAlias)` does not make sense.
+        raise InvalidArgument(f"thing={thing!r} does not make sense as a strategy")
```

Comment:
This is going to be a bit awkward, but I think I'd prefer to write a separate clause for each inline, so that we can have a detailed and specific error message.

I can imagine wanting abstraction or helper functions when we're handling 4+ types like this, but it feels premature now.